service/app/util/Filters.java

```diff
@@ -49,9 +49,12 @@ public Filters(LoggingFilter loggingFilter, GzipFilter filter) {
   // Whether the given request/result should be gzipped or not
   private boolean shouldGzipFunction(RequestHeader v1, ResponseHeader v2) {
     double responseSize = 0.0;
-    if (v2.headers().get(HeaderParam.X_Response_Length.getName()) != null) {
-      String strValue = v2.headers().get(HeaderParam.X_Response_Length.getName()).get();
-      responseSize = Double.parseDouble(strValue);
+    boolean checkResponseLength=v2.headers().contains(HeaderParam.X_Response_Length.getName());
```

Comment:
1.  shouldGzipFunction(RequestHeader v1, ResponseHeader v2) -> change v1 to requestHeader and v2 to responseHeader.
2. checkResponseLength  to responseLengthKeyExist.